lib/core/src/persist/send.rs

```diff
@@ -116,20 +116,36 @@ impl Persister {
         &self,
         from_state: PaymentState,
         to_state: PaymentState,
+        is_local: Option<bool>,
     ) -> Result<()> {
-        let con = self.get_connection()?;
-        con.execute(
-            "UPDATE send_swaps
-            SET
-                state = :to_state
-            WHERE
-                state = :from_state
+        let mut con = self.get_connection()?;
+        let tx = con.transaction_with_behavior(rusqlite::TransactionBehavior::Immediate)?;
+        let mut where_clauses = vec![get_where_clause_state_in(&[from_state])];
+        if let Some(is_local) = is_local {
+            where_clauses.push(format!(
+                "(sync_state.is_local = {} OR sync_state.is_local IS NULL)",
+                is_local as i8
+            ));
+        }
+
+        let ids = self
+            .list_send_swaps_where(&tx, where_clauses)?
+            .iter()
+            .map(|s| s.id.clone())
+            .collect::<Vec<_>>()
+            .join(", ");
```

Comment:
Do we need this extra query or we can pass the where clause directly to the update statement and remove the need for a transaction?

Reply:
Fixed [ac6754d](https://github.com/breez/breez-sdk-liquid/pull/730/commits/ac6754d8e7bee538626e8b99af7bc7798ba4f7ee)